### ISSUE TITLE
Julia jupyter engine errors out

### ISSUE THREAD

**Issue by gyansinha** [quarto-dev/quarto-cli#8067]
### Bug description

Basic document won't render or preview 

### Steps to reproduce

```qmd
---
title: "Untitled"
format: html
jupyter: julia-1.10
---

This is a test.
```

but this will:

```qmd
---
title: "Untitled"
format: html
jupyter: python3
---

This is a test.
```

### Expected behavior

document should render or preview

### Actual behavior

```bash
[xxx@yyy$ quarto preview foo.qmd

Starting julia-1.9 kernel...Traceback (most recent call last):
  File "/home/xxxx/admin/quarto-1.4.533/share/jupyter/jupyter.py", line 21, in <module>
    from notebook import notebook_execute, RestartKernel
  File "/home/xxxx/admin/quarto-1.4.533/share/jupyter/notebook.py", line 14, in <module>
    from yaml import load, BaseLoader
ModuleNotFoundError: No module named 'yaml'
```

### Your environment

```bash
IDE: VSCode
OS: CentOS Stream 9 
```

### Quarto check output

```bash

quarto check

Quarto 1.4.533
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.533
      Path: /home/xxx/admin/quarto-1.4.533/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.09
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /home/xxxx/.TinyTeX/bin/x86_64-linux
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.1 (Conda)
      Path: /home/xxxx/.julia/environments/v1.10/.CondaPkg/env/bin/python
      Jupyter: 5.5.1
      Kernels: julia-_4-threads_-1.9, julia-4-threads-1.9, python3, julia-1.9, julia-1.10

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.1
      Path: /usr/lib64/R
      LibPaths:
        - /home/xxxx/R/x86_64-redhat-linux-gnu-library/4.3
        - /usr/lib64/R/library
        - /usr/share/R/library
      knitr: 1.44
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

**Comment by mcanouil** [comment]
Did you follow <https://quarto.org/docs/computations/julia.html#installation>?

**Comment by gyansinha** [comment]
> Did you follow https://quarto.org/docs/computations/julia.html#installation? Did you try your other Julia versions?
> 
> Julia 1.10 is quite new and it might not yet be supported wel by Jupyter and/or nbconvert.

Yes to the installation question - also tried with Julia 1.9 which has been out for a while with same results.

**Comment by mcanouil** [comment]
Since I cannot reproduce, the issue is with your environment.
So far most of the issues regarding Jupyter setup involves Conda environment.

Could you check your kernel?

```sh
jupyter kernelspec
```

Could you try setting up a clean and fresh Conda/Julia install?
You'll likely need to clenup `~/.julia/`.

**Comment by cderv** [comment]
I see your error is 
````
ModuleNotFoundError: No module named 'yaml'
````

so it means the environment used with julia 1. 9 (`Starting julia-1.9 kernel`) is not configured with all the tools 

I see from your quarto check 
````
[✓] Checking Python 3 installation....OK
      Version: 3.12.1 (Conda)
      Path: /home/xxxx/.julia/environments/v1.10/.CondaPkg/env/bin/python
      Jupyter: 5.5.1
      Kernels: julia-_4-threads_-1.9, julia-4-threads-1.9, python3, julia-1.9, julia-1.10
````

That you are using Conda from Julia environment (at least Quarto finds this one), and you have several julia kernel founds. 

So there could be a lot going on in configuration. 

You can check that your Conda installed python and IJulia have the required package. Possibly just **yaml** is missing. 
See Conda.jl (https://github.com/JuliaPy/Conda.jl) about how to manage modules in this installation. 

If you have also a classic python installation with Jupyter (in addition to the Julia conda one), this could conflict, in the sense that you need to be aware of those different environment to set them up correctly. 

Note that by default, setting up Julia and IJulia will probably use the internal Conda Julia with Conda.jl and you should check this configuration first. (IJulia defaults to this - see details at https://github.com/JuliaLang/IJulia.jl#quick-start)

So overall, this is likely to be a configuration issue. [Troubleshooting](https://quarto.org/docs/troubleshooting/) adives may also help to have more verbose information maybe.

**Comment by gyansinha** [comment]
> I see your error is
> 
> ```
> ModuleNotFoundError: No module named 'yaml'
> ```
> 
> so it means the environment used with julia 1. 9 (`Starting julia-1.9 kernel`) is not configured with all the tools
> 
> I see from your quarto check
> 
> ```
> [✓] Checking Python 3 installation....OK
>       Version: 3.12.1 (Conda)
>       Path: /home/xxxx/.julia/environments/v1.10/.CondaPkg/env/bin/python
>       Jupyter: 5.5.1
>       Kernels: julia-_4-threads_-1.9, julia-4-threads-1.9, python3, julia-1.9, julia-1.10
> ```
> 
> That you are using Conda from Julia environment (at least Quarto finds this one), and you have several julia kernel founds.
> 
> So there could be a lot going on in configuration.
> 
> You can check that your Conda installed python and IJulia have the required package. Possibly just **yaml** is missing. See Conda.jl (https://github.com/JuliaPy/Conda.jl) about how to manage modules in this installation.
> 
> If you have also a classic python installation with Jupyter (in addition to the Julia conda one), this could conflict, in the sense that you need to be aware of those different environment to set them up correctly.
> 
> Note that by default, setting up Julia and IJulia will probably use the internal Conda Julia with Conda.jl and you should check this configuration first. (IJulia defaults to this - see details at https://github.com/JuliaLang/IJulia.jl#quick-start)
> 
> So overall, this is likely to be a configuration issue. [Troubleshooting](https://quarto.org/docs/troubleshooting/) adives may also help to have more verbose information maybe.

Thank you for the tips you provided. As was correctly pointed out, it was a configuration issue at my end for IJulia. After using the 
```
import Conda; Conda.SCRIPTDIR
```
to find out where the IJulia executable lives, I was able to reinstall jupyter and the missing pytaml packages. All works now.

**Comment by cderv** [comment]
> I will just add that I have been using Quarto with Julia for about a year or so. And it stopped working recently. And I am unable to identify the breaking update, but there must have been some. Downgrading to lower versions of Julia within a qmd document does not help. Trying a dev version of Quarto instead of the unstable one does not help either

@hurak If you have a specific error, and specific reproducible example, we'll be happy to look into. You can open a new issue with information. 

Our tests with Julia does not break but we don't test everything... thank you !

**Comment by hurak** [comment]
Thanks for the response. My original motivation was just to enquire the author of the issue about the technical details of his solution that he only sketched here. Upon your encouragement to open a new issue, I indeed started filling in one, but while doing that, I have actually converged to the solution. And I believe that this solution must be identical to the one sketched by the author of the issue. Therefore I will not bother you with a new issue (and sorry for reviving this one).

Just in case anyone bumps across this, Indeed, having several Python installation on my computer, just reinstalling `jupyter` and `pyyaml` packages (through pip) for all of those Python installation helped. I will leave possible identification of which Python is called when for future :-) I am happy now that I made Quarto running with Julia code again. 

And thanks a lot for Quarto.
